pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -319,11 +319,16 @@ private boolean createDictionaryForColumn(ColumnIndexCreationInfo info, SegmentG
       return false;
     }
 
+    // Do not create dictionaries for json or text index columns as they are high-cardinality values almost always
+    if (config.getJsonIndexCreationColumns().contains(column)
```

Comment:
suggest adding tests for both json and text index to ensure no dictionary is created